projects/storefrontlib/src/cms-components/myaccount/forgot-password/forgot-password.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import {
+  AuthConfigService,
+  OAuthFlow,
+  RoutingService,
+  UserService,
+} from '@spartacus/core';
+import { CustomFormValidators } from '../../../shared/utils/validators/custom-form-validators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ForgotPasswordService {
+  constructor(
+    protected fb: FormBuilder,
+    protected userService: UserService,
+    protected routingService: RoutingService,
+    protected authConfigService: AuthConfigService
+  ) {}
+
+  form: FormGroup = new FormGroup({
+    userEmail: new FormControl('', [
+      Validators.required,
+      CustomFormValidators.emailValidator,
+    ]),
+  });
+
+  submit() {
+    if (!this.form.valid) {
+      this.form.markAllAsTouched();
+      return;
+    }
+    this.userService.requestForgotPasswordEmail(this.form.value.userEmail);
+    if (
+      this.authConfigService.getOAuthFlow() ===
+      OAuthFlow.ResourceOwnerPasswordFlow
+    ) {
+      this.routingService.go({ cxRoute: 'login' });
+    }
+  }
+
+  reset() {
```

Comment:
I'd either make the form an internal property (protected) or do a form.reset() in the controller directly.

Reply:
I did it like that because I thought the `reset` method might need to do more stuff in the future not only resetting the form (resetting the state for example).

Reply:
ah, that sounds fair